src/v/rpc/transport.h

```diff
@@ -143,29 +143,46 @@ class transport final : public base_transport {
 };
```

Comment:
to add to what @dotnwat said, i don't see a test that triggers this. we control the stack, so couldn't we inject this in a test.

Reply:
The test is already there, the problem with test was that it did used empty struct as a response and this error wasn't triggered. 

WRT `broken_promise` exception: the broken promise exception was triggered because `streaming_context` was deleted before setting value to the `promise` it contains. The `broken_promise` was just a side effect of `parse_type` throwing an exception. I fixed that behavior in one of the next commits.

Reply:
> to add to what @dotnwat said, i don't see a test that triggers this. we control the stack, so couldn't we inject this in a test.
@senior7515  Here is a commit that fixes test: 
https://github.com/vectorizedio/redpanda/pull/2917/commits/8ae27c0e94b939046a4015b6eb1a4a2517cbcf8b

Reply:
that makes sense thx